File: backend/src/main/java/com/example/backend/Model/Class/Parts.java
package com.example.backend.Model.Class;

import com.example.backend.Model.Enum.PartCategory;
import jakarta.persistence.*;
import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Cacheable
@Getter
@Setter
@Entity
@Table(name = "PARTS")
public class Parts {
    @Id
    private String id;
    @Column(name = "name")
    private String name;
    @Column(name = "category")
    @Enumerated(EnumType.STRING)
    private PartCategory category;
    @Column(name = "unit_cost")
    private Double unit_cost;
}

File: frontend/src/app/_model/_interface/parts.ts
import { PartCategory } from '../_enum/part-category';

export interface Parts {
  id: string;
  name: string;
  category: PartCategory;
  unit_cost: number;
}
export function isParts(obj: any): obj is Parts {
  return (
    obj &&
    typeof obj.id === 'string' &&
    typeof obj.name === 'string' &&
    typeof obj.unit_cost === 'number'
  );
}
